src/fabric_cli/commands/config/fab_config_set.py

```diff
@@ -76,13 +76,64 @@ def _set_config(args: Namespace, key: str, value: Any, verbose: bool = True) ->
 
         Context().cleanup_context_files(cleanup_all_stale=True, cleanup_current=True)
 
-    if (
-        key == fab_constant.FAB_MODE
-        and current_mode == fab_constant.FAB_MODE_COMMANDLINE
-        and previous_mode == fab_constant.FAB_MODE_INTERACTIVE
-    ):
-        utils_ui.print("Exiting interactive mode. Goodbye!")
-        os._exit(0)
+    # Enhanced mode transition handling
+    if key == fab_constant.FAB_MODE:
+        if (current_mode == fab_constant.FAB_MODE_INTERACTIVE
+            and previous_mode == fab_constant.FAB_MODE_COMMANDLINE):
+            # Handle command_line → interactive transition
+            if _is_user_authenticated():
```

Comment:
IMO, we should enter interactive mode whenever we change the config, even if i'm not logged in. the shell mode can act the same as the command_line, i still can use commands like 'config ls', 'pwd', without being authenticated.